modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/VerticalNavTag.java

```diff
@@ -139,16 +169,23 @@ private void _renderVerticalNavItems(
 				(VerticalNavItemList)verticalNavItem.get("items");
 
 			Boolean active = (Boolean)verticalNavItem.get("active");
+			String id = (String)verticalNavItem.get("id");
 
 			if (active == null) {
 				active = Boolean.FALSE;
 			}
+			else if ((_active == null) && (id != null)) {
+				setActive(id);
+			}
```

Comment:
Yeah I understand that concern. I don't like the duality between tag attribute and internal set logic. The alternative I see is to deal with it in the `prepareProps()` and have a policy of what has more priority. That may as well happen in the getter. I'm fine redefining this

Reply:
it doesn't seem to return false

<img width="914" alt="image" src="https://github.com/liferay-frontend/liferay-portal/assets/46778114/b904a3de-2da6-401a-b7ba-006e519614de">

Reply:
OK, then we need to do a bit more complex expression, something like
```java
if ((id != null) && (Objects.equals(id, _active))) {
    ...
}
```

Reply:
> This does not look right. It feels wrong to setting the attribute value in the rendering logic.

Sorry this is pretty confusing for me, I'm not sure I understood what you are asking for, so I'm going to try to split your message, let me know if I'm understanding it right.

> This does not look right. It feels wrong to setting the attribute value in the rendering logic.

Should I move the `setActive(itemId)` outside the `private void _renderVerticalNavItems(` and put it inside the `protected int processStartTag() throws Exception` ?

> If the setter has some custom logic, it could in theory be inside the setter, but the standard pattern in DXP is to have vanilla setter and complex getter if needed.

By custom logic do you mean the if/checks/conditional we have on the variables?

> Again in theory, we could loop over items in `getActive()`. But I don't think any of that is needed. Clay handles this case with tag attribute unset, and we only need internal flag variable to add CSS classed in SSR.

Should I remove the part of getting the "active" (that we are going to pass to react) from the verticalNavItems? (And leave it without "active" unless defined in the parent component?)

> Additional problem in above code is that the item attribute has higher priority: if it's unset we call it false and don't even check tag attribute.

Should I check the `_active` property from the parent before the `itemActive` property from the `verticalNavItems` ?

> 
> To fix both above issues, we can have something like:
> 
> ```java
> if (Objects.equals(_active, id)) {
>     itemActive = true;
> }

I tested and if both are `false` it will return `true` should I change it to something like:

```java
if (_active && (_active == id)) {
     itemActive = true;
}
```

?

plus we need to declare its type before defining it, right?

```java
if (_active && (_active == id)) {
    Boolean itemActive = Boolean.TRUE;
}
```

additional question: is `true` and `Boolean.TRUE` tha same?

> else {
>     Boolean itemActive = (Boolean)verticalNavItem.get("active");
> 
>     if (itemActive == null) {
>         itemActive = false;
>     }

If we needed to preserve the part of getting the "active" (that we are going to pass to react) from the verticalNavItems we might need to repeat the check on the `id` variable since we never tested in this conditional, right?

> }
> ...
> if (itemActive) {...}
> ```
> 
> Notes: I am not sure if Objects.equals(false, false) returns false. I'm assuming it does.